tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,31 @@
+"""Fill in the Gaps."""
+import os
+
+
+def fill_gaps(folder_path, prefix):
+    """Fill in the Gaps."""
+    # init file list with first not needed element
+    file_list = [None]
+    os.chdir(folder_path)
+    # get list of tuples of filenames and extentions
+    with os.scandir(folder_path) as scandir:
+        for entry in scandir:
+            if entry.name.startswith(prefix) and entry.is_file():
+                file_list.append(os.path.splitext(entry.name))
+    # check if sequence number equal to number in filename and rename it if is not.
+    number_of_zeroes = len(file_list[1][0]) - len(prefix)
```

Comment:
Nice try to have dynamic length of number. However, this approach has some downsides:

- You are getting second file from list by `[1]`. What if there are no files at all?
- What if second file does not have most zeros?

Maybe better approach is to set `number_of_zeros` according to number of files. Simplest approach is
`number_of_zeros = len(str(len(file_list)))`. It looks a bit tricky, but if basically if you have tens of files, then number of zeros would 2, if you have hundreds of files it would be 3 and so on. Going further you may want to introduce minimal amount of zeroes:
```python
MIN_ZEROES = 3
number_of_zeroes = max(MIN_ZEROES, len(str(len(file_list))))
```

Reply:
I would say that task description is very on high level and has a lot of unsaid and non clarified parts. 
I made it as i got it. 
But I agree it can be done in other way.

Reply:
Yeah, it is totally optional ;)